Source/WebCore/platform/graphics/skia/FontPlatformDataSkia.cpp

```diff
@@ -54,6 +54,42 @@ FontPlatformData::FontPlatformData(sk_sp<SkTypeface>&& typeface, float size, boo
     m_features = WTFMove(features);
 }
 
+FontPlatformData::FontPlatformData(float size, FontOrientation&& orientation, FontWidthVariant&& widthVariant, TextRenderingMode&& textRenderingMode, bool syntheticBold, bool syntheticOblique, RefPtr<FontCustomPlatformData>&& customPlatformData)
+    : FontPlatformData(size, syntheticBold, syntheticOblique, orientation, widthVariant, textRenderingMode, customPlatformData.get())
+{
+    m_font = SkFont(customPlatformData->m_typeface, m_size);
+    m_font.setEmbolden(m_syntheticBold);
+    m_font.setSkewX(m_syntheticOblique ? -SK_Scalar1 / 4 : 0);
+    m_font.setEdging(FontRenderOptions::singleton().antialias());
+    m_font.setHinting(FontRenderOptions::singleton().hinting());
+
+    m_hbFont = SkiaHarfBuzzFont::getOrCreate(*m_font.getTypeface());
+}
+
+std::optional<FontPlatformData> FontPlatformData::fromIPCData(float size, FontOrientation&& orientation, FontWidthVariant&& widthVariant, TextRenderingMode&& textRenderingMode, bool syntheticBold, bool syntheticOblique, IPCData&& ipcData)
+{
+    if (ipcData.fontFaceData.isEmpty())
+        return std::nullopt;
+
+    auto fontFaceData = SharedBuffer::create(WTFMove(ipcData.fontFaceData));
+    RefPtr fontCustomPlatformData = FontCustomPlatformData::create(fontFaceData, ipcData.itemInCollection);
+    if (!fontCustomPlatformData)
+        return std::nullopt;
+
+    return FontPlatformData(size, WTFMove(orientation), WTFMove(widthVariant), WTFMove(textRenderingMode), syntheticBold, syntheticOblique, WTFMove(fontCustomPlatformData));
+}
+
+FontPlatformSerializedCreationData FontPlatformData::toIPCData() const
+{
+    const auto& data = creationData();
+    if (data)
+        return { { data->fontFaceData->span() }, data->itemInCollection };
+
+    auto serialized = m_font.getTypeface()->serialize();
+    std::span<const uint8_t> copy(serialized->bytes(), serialized->size());
+    return { WTFMove(copy), ""_s };
```

Comment:
So, we serialize the font, but it's encoded as if it was FontCustomPlatformData? I think we could add encoder/decode for sk_sp<SkTypeface> similar to the one we added to color spaces and here use a variant, to either use the data for FontCustomPlatformData or the font for FontPlatformData

Reply:
Researching this idea.

Reply:
@achristensen07 noticed that this is going to leak because a `std::span` is just a pointer and size which has no ownership.

Addressing in https://github.com/WebKit/WebKit/pull/28301 and then will follow up with changes here that resemble what you were talking about.

Reply:
@donny-dont I don't see any leak here? We construct a Vector from the span. The Vector copies the span data and keeps its memory alive. I don't think we need to free the value returned by serialize() either since it is returning a pointer.

Reply:
We should probably never name a `std::span` "copy" since that's not what it does?

Reply:
> We should probably never name a `std::span` "copy" since that's not what it does?

That is very true :)

Reply:
So this was changed to be a variant as requested